### ISSUE TITLE
pyarrow broken in pyspark-notebook

### ISSUE THREAD

**Issue by tschm** [jupyter/docker-stacks#944]
pyarrow is broken in the pyspark-notebook where it is installed

docker run -it --rm -p 8888:8888 jupyter/pyspark-notebook:latest

Start a Python notebook

try:
```
import pyarrow as pa

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-852643f3aad4> in <module>
----> 1 import pyarrow as pa

/opt/conda/lib/python3.7/site-packages/pyarrow/__init__.py in <module>
     52 
     53 
---> 54 from pyarrow.lib import cpu_count, set_cpu_count
     55 from pyarrow.lib import (null, bool_,
     56                          int8, int16, int32, int64,

ImportError: /opt/conda/lib/python3.7/site-packages/pyarrow/lib.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZNK5arrow6Status8ToStringB5cxx11Ev
```
Version 0.11.1 of pyarrow is installed in pyspark. This problem of pyarrow is well-known in the arrow community. Trying to update pyarrow in my dockerfile results in some conflicts:

```
UnsatisfiableError: The following specifications were found to be incompatible with each other:



Package libprotobuf conflicts for:
protobuf=3.9 -> libprotobuf[version='3.9.0.*,3.9.1.*,3.9.2.*,>=3.9.0,<3.10.0a0,>=3.9.1,<3.10.0a0,>=3.9.2,<3.10.0a0']
libprotobuf

ERROR: Service 'jupyter' failed to build: The command '/bin/sh -c conda install -y -c conda-forge pyarrow=0.13.0 arrow-cpp parquet-cpp' returned a non-zero code: 1
/home/thomas/github/docker/jupyter_classic/Makefile:23: recipe for target 'build' failed
make: *** [build] Error 1


```

**Comment by rberenguel** [comment]
This won't fix the issue overall, but at least for working around it in a notebook:

```
import sys
!{sys.executable} -m pip uninstall pyarrow --yes
!{sys.executable} -m pip install pyarrow
```

and then restart the kernel. This will install 0.15 by the way, it was released recently.

**Comment by Bidek56** [comment]
This seem to be a Dockerfile fix: `RUN conda install -c conda-forge --quiet -y arrow-cpp=0.12.* pyarrow`
But because of [Conda issue](https://github.com/conda/conda/issues/9367), PR won't build for now.

**Comment by romainx** [comment]
The version of `pyarrow` has been updated to `0.16` in the latest `pyspark-notebook` [version](https://github.com/jupyter/docker-stacks/wiki/pyspark-notebook-94fdd01b492f).

```python
import pyarrow as pa
pa.__version__

# '0.16.0'
```

So if everyone agree, we could close this issue.

However, and more generally I will work on a way to avoid this kind of issues in the future by improving the tests.

Best

**Comment by romainx** [comment]
@tschm I'm going to close this issue since it's fixed now with `pyarrow` version `0.16`. Feel free to open a new issue if you encounter this kind of problem again. 

For information we are working on a way to test the import of the packages in the PR #1037 in order to avoid this kind of obvious problem in the future.
